contracts/src/bridge/Inbox.sol

```diff
@@ -410,7 +410,11 @@ contract Inbox is DelegateCallAware, PausableUpgradeable, IInbox {
         bytes calldata data
     ) external payable whenNotPaused onlyAllowed returns (uint256) {
         // ensure the user's deposit alone will make submission succeed
-        if (msg.value < (maxSubmissionCost + l2CallValue + gasLimit * maxFeePerGas)) {
+        // certain legacy caller can bypass this check and revert to the unsafe behaviour
+        if (
+            msg.value < (maxSubmissionCost + l2CallValue + gasLimit * maxFeePerGas) ||
+            msg.sender == UNSAFE_CREATERETRYABLETICKET_CALLER
```

Comment:
I think this check is backwards. Currently it'll always revert if called by `UNSAFE_CREATERETRYABLETICKET_CALLER` it seems

Reply:
```suggestion
            msg.sender != UNSAFE_CREATERETRYABLETICKET_CALLER
```

Reply:
fixed